app/javascript/find/controllers/locations_autocomplete_controller.js

```diff
@@ -1,60 +1,4 @@
-import { Controller } from '@hotwired/stimulus'
-import accessibleAutocomplete from 'accessible-autocomplete'
-import debounce from 'lodash.debounce'
-import { request } from '../../utils/request_helper'
+import RemoteAutocompleteController from './remote_autocomplete_controller'
 
-export default class extends Controller {
-  static targets = ['input']
-  static values = { path: String, selectNameAndCode: Boolean }
-
-  connect () {
-    if (!this.hasInputTarget) return
-
-    const inputElement = this.inputTarget
-
-    accessibleAutocomplete({
-      element: inputElement.parentElement,
-      id: inputElement.id,
-      showNoOptionsFound: false,
-      name: inputElement.name,
-      defaultValue: inputElement.value,
-      displayMenu: 'overlay',
-      minLength: 3,
-      source: this.fetchSuggestions(),
-      templates: {
-        inputValue: this.suggestionTemplate,
-        suggestion: this.suggestionTemplate
-      },
-      confirmOnBlur: false
-    })
-
-    inputElement.remove()
-  }
-
-  fetchSuggestions () {
-    return debounce((query, populateResults) => {
-      if (!query) return populateResults([])
-
-      new Promise((resolve, reject) => {
-        request(this.pathValue)(query, (data) => {
-          resolve(data)
-        })
-      })
-        .then((data) => {
-          const suggestions = Array.isArray(data?.suggestions) ? data.suggestions : []
-          populateResults(suggestions)
-        })
-        .catch(() => {
-          populateResults([])
-        })
-    }, 900)
-  }
-
-  suggestionTemplate (result) {
-    if (typeof result === 'string') {
-      return result
-    }
-
-    return result && typeof result.name === 'string' ? result.name : ''
-  }
+export default class extends RemoteAutocompleteController {
```

Comment:
Empty but just open for extension.

See https://github.com/DFE-Digital/publish-teacher-training/pull/4935/files#r1961372936